### PR TITLE
Fix ValueGeneratedOnUpdate in scaffolding

### LINKED ISSUE #18579: Scaffolding column with `ValueGenerated.OnUpdate` generates invalid C# code
See https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/issues/877 for an example (issue 2).

The issue is, that the following `switch` is missing the handling of `ValueGenerated.OnUpdate`:

https://github.com/aspnet/EntityFrameworkCore/blob/27c8d56284a641ef0836972ced08db8f7433674d/src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs#L701-L730

It will therefore generate `entity.Property(e => e.MyProperty).()` for properties with `ValueGenerated.OnUpdate`.

Is that an oversight or a deliberate choice?

### PR DESCRIPTION
**Pull request by roji**
And add some tests.

Fixes #18579

### CODE REVIEW

File: src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs [hunk 1]
```diff
@@ -703,28 +703,16 @@ private void GenerateProperty(IProperty property, bool useDataAnnotations)
             if (((IConventionProperty)property).GetValueGeneratedConfigurationSource().HasValue
                 && RelationalValueGenerationConvention.GetValueGenerated(property) != valueGenerated)
             {
-                string methodName;
-                switch (valueGenerated)
+                var methodName = valueGenerated switch
```
Comment (smitpatel) [inline src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs]: switch block.

File: src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs [hunk 2]
```diff
@@ -703,28 +703,16 @@ private void GenerateProperty(IProperty property, bool useDataAnnotations)
...
-                        methodName = "";
-                        break;
-                }
+                    ValueGenerated.OnAdd => nameof(PropertyBuilder.ValueGeneratedOnAdd),
+                    ValueGenerated.OnAddOrUpdate => property.IsConcurrencyToken
+                        ? nameof(PropertyBuilder.IsRowVersion)
+                        : nameof(PropertyBuilder.ValueGeneratedOnAddOrUpdate),
+                    ValueGenerated.OnUpdate => nameof(PropertyBuilder.ValueGeneratedOnUpdate),
+                    ValueGenerated.Never => nameof(PropertyBuilder.ValueGeneratedNever),
+                    _ => throw new NotImplementedException()
```
Comment (smitpatel) [inline src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs]: InvalidOperationException
Add exception message.

Comment (roji) [reply src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs]: No problem - note we have quite a few switch/case throwing NotImplementedException around the code, should clean up at some point.

File: src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs (hunk 1 above)
Comment (roji) [reply src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs]: Any reason here? We agreed to use it except for complex situations, and this seems as simple as it gets. If we don't use it here we basically don't use it anywhere, no?

Comment (smitpatel) [reply src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs]: 3 line ternary as a return of one of the statement. That is complex enough.

Comment (roji) [reply src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs]: This is not how I understand the decision made by the team, and basically invalidates the use of this construct for the vast majority of cases. This isn't extremely important but maybe another opinion would help clarify.

Comment (smitpatel) [reply src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs]: @ajcvickers

Comment (ajcvickers) [reply src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs]: Seems fine to me. Still readable.

**Pull request merged**